code/cvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -383,12 +417,12 @@ impl OrderContract<'_> {
 				.sum();
 			let b_total_in: u128 = alternative_all_orders
 				.iter()
-				.filter(|x| x.given().denom == b)
+				.filter(|x: &&SolvedOrder| x.given().denom == b)
 				.map(|x| x.given().amount.u128())
 				.sum();
-			// so do all cows up to bank
+
 			let alternative_transfers = solves_cows_via_bank(
-				alternative_all_orders.clone(),
+				&alternative_all_orders.clone(),
```

Comment:
Is clone necessary here?
```suggestion
				&alternative_all_orders,
```